src/Parsing/String/Basic.purs

```diff
@@ -120,7 +120,7 @@ number =
 numberRegex :: forall m. ParserT String m String
 numberRegex = either unsafeCrashWith identity $ regex pattern mempty
   where
-  pattern = "[+-]?[0-9]*(\\.[0-9]*)?([eE][+-]?[0-9]*(\\.[0-9]*))?"
+  pattern = "[+-]?[0-9]*(\\.[0-9]*)?([eE][+-]?[0-9]*(\\.[0-9]*)?)?"
```

Comment:
```
[+-]? -- these chars are optional
[0-9]* -- there can be 0 or more of these chars
(\\.[0-9]*)? -- these chars can be optional
(
  [eE] -- not optional
  [+-]? -- optional
  [0-9]* -- 0 or more chars
  (\\.[0-9]*)? -- now optional (previously wasn't)
)? -- Wait! This entire block is already optional!?
```
Nevermind, I think my issue is with the regex as a whole since all parts are optional.

Reply:
Huh, that is a good point. There should probably be an "or" somewhere early on so that it matches either a digit or a `.` followed by a digit at a minimum.

Reply:
It's really hard to write a regex that admits all legal number strings but no other strings, but luckily that's not what we're doing here. We only need to find the number substring boundary so that we can pass the substring to the `Data.Number.fromString` function. So

1. We need to admit all number strings. That's the first importance, and this PR fixes that.
2. It's okay if we admit some string that are not number strings. They'll be rejected by `Data.Number.fromString`.

Reply:
Ah okay, that totally makes sense! I probably should have looked outside the context of this regex.